### PR TITLE
Merge dev giddh 2.0

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by aditi-tiwari17 — APPROVED** [review]

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 1]
```diff
@@ -6176,4 +6185,36 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
         }
     }
+
+    /**
+     * To get tax list and assign values to local variables
+     *
+     * @memberof ProformaInvoiceComponent
+     */
+    public getAllDiscounts(): void {
+        this.store.pipe(select(p => p.company.isGetTaxesSuccess), takeUntil(this.destroyed$)).subscribe(isGetTaxes => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:6195]: avoid single character variable

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html [hunk 2]
```diff
@@ -450,6 +450,42 @@ <h3>Address {{i+1}}</h3>
...
 
     </div>
+    <div class="form" *ngIf="isDebtorCreditor && isOtherSelectedTab">
+        <div class="mrB2 selectBox">
+            <div class="form-group ">
+                <label>Default Discount <span
+                    class="primary_clr"></span></label>
+                <div class="clearfix">
+                <span>
+                        <sh-select class="pull-left" [options]="discountList" [isFilterEnabled]="true" [forceClearReactive]="forceClearDiscount$ | async"  [multiple]="true" [(ngModel)]="selectedDiscounts" [placeholder]="'Select discounts'" style="width: calc(100% - 93px);"></sh-select>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html:460]: avoid inline css

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 3]
```diff
@@ -1435,4 +1443,26 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+/**
+ * To apply discount in accounts
+ *
+ * @memberof AccountUpdateNewDetailsComponent
+ */
+public applyDiscounts(): void {
+        if (this.accountDetails) {
+            this.activeAccountName = this.accountDetails.uniqueName;
+        } else {
+            this.activeAccount$.pipe(take(1)).subscribe(a => this.activeAccountName = a.uniqueName);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1456]: avoid single character variable

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html [hunk 4]
```diff
@@ -145,13 +135,49 @@
...
 </div>
+
+<div class="horizontal-form mr1" *ngIf="isDebtorCreditorGroups">
+    <div class="mrB2 selectBox">
+        <div class="form-group ">
+            <label>Default Discounts <span
+                        class="primary_clr"></span></label>
+            <div class="clearfix">
+                <span>
+                        <sh-select class="pull-left" [options]="discountList" [isFilterEnabled]="true" [forceClearReactive]="forceClearDiscount$ | async"  [multiple]="true" [(ngModel)]="selectedDiscounts" [placeholder]="'Select discounts'" style="width: calc(100% - 93px);"></sh-select>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html:146]: remove inline css

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts [hunk 5]
```diff
@@ -72,13 +75,24 @@ export class GroupUpdateComponent implements OnInit, OnDestroy, AfterViewInit {
...
     public showTaxes: boolean = false;
+    // list of groups with less details (parent information)
+    public groupWithParentLessDetailsList:any[] = [];
     @ViewChild('deleteGroupModal', {static: true}) public deleteGroupModal: ModalDirective;
     @ViewChild('moveToGroupDropDown', {static: true}) public moveToGroupDropDown: ShSelectComponent;
+    /** To check is groups belongs to debtor or creditors type  */
+    public isDebtorCreditorGroups: boolean = false;
+    /** To check discount box show/hide */
+    public showDiscount: boolean = false;
+    public discountList: any[] = [];
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts:86]: add JS doc

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts [hunk 6]
```diff
@@ -514,11 +540,65 @@ export class GroupUpdateComponent implements OnInit, OnDestroy, AfterViewInit {
...
+    * @param {*} item
+    * @memberof GroupUpdateComponent
+    */
+    public isDebtorCreditorGroup(itemUniqueName: any): boolean {
+        let isTaxableGroup: boolean = false;
+        const item = this.groupWithParentLessDetailsList.find(element => (itemUniqueName === element.uniqueName));
+        if (item) {
+            isTaxableGroup = item.parentGroups.some(groupName => groupName.uniqueName === 'sundrydebtors' || groupName.uniqueName === 'sundrycreditors');
+        }
+        // this.groupWithParentLessDetailsList.forEach(element => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts:555]: remove commented code if not required

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 7]
```diff
@@ -5781,6 +5772,24 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+            });
+        }
+        if (data && data.applicableDiscounts && data.applicableDiscounts.length) {
+            this.accountAssignedApplicableDiscounts = [...data.applicableDiscounts];
+            if (data.inheritedDiscounts && data.inheritedDiscounts.length) {
+                data.inheritedDiscounts.forEach(element => {
+                    this.accountAssignedApplicableDiscounts.push(...element.applicableDiscounts)
+                });
+            }
+            console.log('testing a/c discount', this.accountAssignedApplicableDiscounts);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:5790]: remove console

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 8]
```diff
@@ -1435,4 +1442,26 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+    public applyDiscounts(): void {
+        if (this.accountDetails) {
+            this.activeAccountName = this.accountDetails.uniqueName;
+        } else {
+            this.activeAccount$.pipe(take(1)).subscribe(activeAccountState => this.activeAccountName = activeAccountState.uniqueName);
+        }
+        if (this.activeAccountName) {
+            _.uniq(this.selectedDiscounts);
+            let assignDescountObject: ApplyDiscountRequestV2 = new ApplyDiscountRequestV2();
+            assignDescountObject.uniqueName = this.activeAccountName;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1460]: fix discount spelling

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts [hunk 9]
```diff
@@ -514,11 +540,65 @@ export class GroupUpdateComponent implements OnInit, OnDestroy, AfterViewInit {
...
+     * @memberof GroupUpdateComponent
+     */
+    public applyDiscounts(): void {
+        let activeGroupUniqueName: string;
+        this.activeGroup$.pipe(take(1)).subscribe(grp => activeGroupUniqueName = grp.uniqueName);
+
+        if (activeGroupUniqueName) {
+            _.uniq(this.selectedDiscounts);
+            let assignDescountObject: ApplyDiscountRequestV2 = new ApplyDiscountRequestV2();
+            assignDescountObject.uniqueName = this.uniqueName;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts:575]: fix discount spelling

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/import-excel/upload-file/upload-file.component.ts [hunk 10]
```diff
@@ -56,7 +65,26 @@ export class UploadFileComponent implements OnInit {
...
 
+    /**
+     * Initializes the component
+     *
+     * @memberof UploadFileComponent
+     */
+
     public ngOnInit(): void {
+        this.activatedRoute.params.pipe(takeUntil(this.destroyed$)).subscribe(data => {
+            this.entity = data.type;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/import-excel/upload-file/upload-file.component.ts:76]: nullity check for data variable

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 11]
```diff
@@ -5781,6 +5772,24 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+            console.log('testing a/c tax', this.tcsTdsTaxesAccount);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:5791]: remove console

**Review by rohit-walkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts (hunk 1 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:6195]: Existing code was moved to a method by @rohit-walkover

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html (hunk 2 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.html]: This is commented now in latest code

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts (hunk 3 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: This is commented now in latest code

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html (hunk 4 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html]: Done

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts (hunk 5 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts:86]: Done

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts (hunk 8 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1460]: Done

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts (hunk 9 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.ts:575]: Done

File: apps/web-giddh/src/app/import-excel/upload-file/upload-file.component.ts (hunk 10 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/import-excel/upload-file/upload-file.component.ts:76]: Done

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts (hunk 11 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:5791]: Done :)

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**